File: Javascript/HappyNumber.js
// Leetcode Problem - 202

var isHappy = function (n) {
  const hashSet = new Set();

  while (n !== 1 && !hashSet.has(n)) {
    hashSet.add(n)
    const digits = Array.from(String(n), Number);
    
    let sum = 0;
    for (element of digits) {
      sum += Math.pow(element, 2);
    }

    n = sum;
  }

  if (n === 1) {
    return true;
  } else {
    return false;
  }
};

const n = 2;
console.log(isHappy(n));

File: Python/HappyNumber.py
#Leetcode Problem - 202

def isHappy(n):
    hashSet = set()
    

    while n!=1 and n not in hashSet:
        hashSet.add(n)
        digits = [int(digit) for digit in str(n)]
        sum = 0
        for element in digits:
            sum += element**2

        n = sum

    if n==1:
        return True
    elif n in hashSet:
        return False

n = 2
print(isHappy(n))